File: Golang/ThreeSumClosest/ThreeSumClosest_test.go
package ThreeSumClosest

import "testing"

func TestThreeSumClosest(t *testing.T) {

	if threeSumClosest([]int{-1, 2, 1, 4}, 1) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if threeSumClosest([]int{1, 1, 1, 0}, 100) == 3 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if threeSumClosest([]int{1, 1, 1, 0}, -100) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if threeSumClosest([]int{-1, 2, 1, 4}, 1) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/ThreeSumClosestTest.py
from unittest import TestCase

from ThreeSumClosest import ThreeSumClosest


class TestThreeSumClosest(TestCase):
    def test_threeSumClosest(self):
        t = ThreeSumClosest()

        self.assertTrue(t.threeSumClosest([-1, 2, 1, 4], 1) == 2)
        
        self.assertTrue(t.threeSumClosest([1, 1, 1, 0], 100) == 3)
        
        self.assertTrue(t.threeSumClosest([1, 1, 1, 0], -100) == 2)
        
        self.assertTrue(t.threeSumClosest([-1, 2, 1, -4], 1) == 2)
